File: javascript/darts/darts.js
const getHypotenuse = (x, y) => Math.sqrt(Math.abs(x) ** 2 + Math.abs(y) ** 2);

export const score = (x, y) => {
  let score = 0;
  const hypotenuse = getHypotenuse(x, y);

  if (hypotenuse <= 1) {
    score = 10;
  } else if (hypotenuse <= 5) {
    score = 5;
  } else if (hypotenuse <= 10) {
    score = 1;
  }

  return score;
};

File: ruby/darts/darts.rb
class Darts
  attr_reader :radius

  def initialize(x, y)
    @radius = Math.sqrt(x**2 + y**2)
  end

  def score
    if radius <= 1
      10
    elsif radius <= 5
      5
    elsif radius <= 10
      1
    else
      0
    end
  end
end
